adserver/tasks.py

```diff
@@ -759,6 +762,50 @@ def notify_on_ad_image_change(advertisement_id):
     )
 
 
+@app.task()
+def notify_of_autorenewing_flights(days_before=7):
+    """Send a note to flights set to renew automatically."""
+    # Flight must end in exactly `days_before` days
+    # to receive the notification
+    end_date = get_ad_day().date() + datetime.timedelta(days=days_before)
+
+    for flight in Flight.objects.filter(
+        live=True,
+        auto_renew=True,
+        end_date=end_date,
+    ).select_related():
+        log.info("Notifying about flight %s auto-renewing", flight)
+        if settings.FRONT_ENABLED:
+            advertiser = flight.campaign.advertiser
```

Comment:
I've been wondering if we should create properties to make these types of access a bit easier. I always forget the chain of models to get the data I want. Though it does hide queries, but we could do select_related smartly in the property when it nests more than 1 deep 🤔

Reply:
I don't think 2 properties vs. 1 is a big difference to be honest and this is more explicit.